File: RagnarokBotWeb/ClientApp/src/app/models/custom-task.dto.ts
export class CustomTaskDto {
    id: number;
    name: string;
    description?: string;
    enabled: boolean;
    isBlockPurchaseRaidTime: boolean;
    startMessage?: string;
    minPlayerOnline?: number;
    lastRunned?: Date;
    taskType: number;
    scumServerId: number;
    commands: string;
    cron: string;
}

File: RagnarokBotWeb/Domain/Services/Dto/CustomTaskDto.cs
using RagnarokBotWeb.Domain.Enums;

namespace RagnarokBotWeb.Domain.Services.Dto
{
    public class CustomTaskDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public bool Enabled { get; set; }
        public string Cron { get; set; }
        public bool IsBlockPurchaseRaidTime { get; set; }
        public string? StartMessage { get; set; }
        public long? MinPlayerOnline { get; set; }
        public DateTime? LastRunned { get; set; }
        public ECustomTaskType TaskType { get; set; }
        public long? ScumServerId { get; set; }
        public string? Commands { get; set; }
    }
}
